### PR TITLE
Add Tracking docs

### LINKED ISSUE #71: Write documentation for tracking component
We need to write the documentation for the tracking component.

https://mautic-developer.readthedocs.io/en/latest/components/tracking_script.html

https://developer.mautic.org/#manipulating-contacts-(leads)

https://developer.mautic.org/#hooking-into-the-tracking-process-and-returning-custom-responses

### PR DESCRIPTION
**Pull request by dennisameling**
Closes #71 

As requested in https://github.com/mautic/developer-documentation-new/issues/71, this PR ports all tracking-related docs, both on Contacts and for MauticJS (`mtc.js`):

https://developer.mautic.org/#manipulating-contacts-(leads)
https://developer.mautic.org/#hooking-into-the-tracking-process-and-returning-custom-responses

<a href="https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/95"></a>

### CODE REVIEW

File: docs/components/contacts.rst [hunk 1]
```diff
@@ -4,8 +4,170 @@ Contacts
...
+
+  use Mautic\CoreBundle\Helper\IpLookupHelper;
+  use Mautic\LeadBundle\Entity\Lead;
+  use Mautic\LeadBundle\Entity\LeadRepository;
+  use Mautic\LeadBundle\Model\FieldModel;
+  use Mautic\LeadBundle\Model\LeadModel;
+  use Mautic\LeadBundle\Tracker\ContactTracker;
+
+  class ContactService
+  {
```
Comment (dennisameling) [inline docs/components/contacts.rst:35]: The [code block in the old docs](https://developer.mautic.org/#manipulating-contacts-(leads)) was severely outdated and referencing methods that no longer work. The code block below doesn't contain any non-working code, though it does reference some deprecated methods like ``getUniqueIdentiferFields()``. This should be "good enough for now".

File: docs/components/contacts.rst [hunk 2]
```diff
@@ -4,8 +4,170 @@ Contacts
...
+
+  declare(strict_types=1);
+
+  namespace MauticPlugin\HelloWorldBundle\Services;
+
+  use Mautic\LeadBundle\Entity\Lead;
+  use Mautic\LeadBundle\Tracker\ContactTracker;
+
+  class ContactTrackingService
+  {
```
Comment (dennisameling) [inline docs/components/contacts.rst:146]: Also refactored this code sample. [Old docs here](https://developer.mautic.org/#manipulating-contacts-(leads)).

File: docs/components/tracking_script.rst [hunk 3]
```diff
@@ -1,8 +1,442 @@
...
\ No newline at end of file
+.. vale off
+
+Tracking script MauticJS
+########################
+
+.. vale on
+
+Mautic provides a means for Plugins to inject custom JavaScript into ``mtc.js``, the PHP generated script that manages Mautic's tracking pixel, Dynamic Web Content, etc.
+You can embed ``mtc.js`` in third party websites to manage communication between those and Mautic.
```
Comment (dennisameling) [inline docs/components/tracking_script.rst:9]: The only thing I did for the `MauticJS` docs is to port everything over to RST and fix Vale-related errors/warnings/suggestions. I haven't actually tested the code samples, but `MauticJS` doesn't really change a lot so most of it (if not everything) should still work.

File: docs/components/tracking_script.rst [hunk 4]
```diff
@@ -1,8 +1,438 @@
...
+.. code-block:: js
+
+   MauticFormCallback['replaceWithFormName'] = {
+       onResponseEnd: function (response) {
+            // called to process the response to the form submission
+       },
+   };
+
+Called at the end default Form submission response processing. Receives ``response`` containing the Form submission response.
+Return value isn't required and not processed.
```
Comment (RCheesley) [inline docs/components/tracking_script.rst:237]:
```suggestion
Return value isn't required and isn't processed.
```

File: docs/components/tracking_script.rst [hunk 5]
```diff
@@ -1,8 +1,438 @@
...
+
+   MauticFormCallback['replaceWithFormName'] = {
+       onShowPreviousPage: function (pageNumber) {
+            // called prior to going back to previous page
+       },
+   };
+
+.. vale off
+
+MauticJS API Functions
```
Comment (RCheesley) [inline docs/components/tracking_script.rst:323]:
```suggestion
MauticJS API functions
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Thanks for the PR @dennisameling - a few suggestions but please do check them for technical validity as I lack the contextual understanding for some of these.

File: docs/components/contacts.rst [hunk 6]
```diff
@@ -4,8 +4,170 @@ Contacts
...
+          $this->leadModel      = $leadModel;
+          $this->contactTracker = $contactTracker;
+          $this->ipLookupHelper = $ipLookupHelper;
+          $this->fieldModel     = $fieldModel;
+          $this->leadRepository = $leadRepository;
+      }
+
+      public function createLead()
+      {
+          // Currently tracked lead based on cookies
```
[the remark this answers is not in the document]
Comment (RCheesley) [reply docs/components/contacts.rst:58]: While we can't change the references in code, we can update the references in the comments to use the more updated terminology. WDYT?

File: docs/components/tracking_script.rst [hunk 7]
```diff
@@ -1,8 +1,438 @@
...
+   MauticFormCallback['replaceWithFormName'] = {
+       onValidateEnd: function (formValid) {
+            // before form submit
+       },
+       onResponse: function (response) { 
+            // after form submit
+       }
+   };
+
+If you wish execute additional code to execute as the Form is getting processed, create a ``MauticFormCallback`` object.
```
[the remark this answers is not in the document]
Comment (dennisameling) [reply docs/components/tracking_script.rst:101]: `being processed` is not allowed by Vale as it's passive voice. Changed it to `If you wish to run additional code before or after submission of the Form, create a ``MauticFormCallback`` object.`

[the remark this answers is not in the document]
Comment (RCheesley) [reply docs/components/tracking_script.rst]: Ahh, lesson to review PRs within the environment to check for the Vale errors! Thanks!

**Pull request merged**